PathPlanning/RRTStar/rrt_star.py

```diff
@@ -161,7 +218,11 @@ def rewire(self, new_node, near_inds):
             improved_cost = near_node.cost > edge_node.cost
 
             if no_collision and improved_cost:
-                self.node_list[i] = edge_node
+                near_node.x = edge_node.x
```

Comment:
This change is for #383. Do you think just updating edge_node.parent to near_node one works? For example:

edge_node.parent = near_node.parent
self.node_list[i] = edge_node

Reply:
The problem is that there could be a single node such that its parent attribute contains a reference to near_node. The line `self.node_list[i] = edge_node` does not update the children of `self.node_list[i]`. So, if we don't want to look up for the children of `self.node_list[i]` in order to update their parent attribute, we have to update the attributes of instance referenced by its children's parent attribute.

Reply:
OK. Make sense.